src/combinatorics/simple_graph/triangle/basic.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.simple_graph.clique
+
+/-!
+# Triangles in graphs
+
+This file defines triangles in simple graphs.
+
+A triangle is a `3`-clique, that is a set of three vertices which are all connected.
+
+## Main declarations
+
+* `simple_graph.triangle_finset`: Finset of triangles of a graph.
+* `simple_graph.triangle_free`: Predicate for a graph to have no triangle.
+* `simple_graph.triangle_free_far`: Predicate for a graph to have enough triangles that one must
+  remove a lot of edges to remove all of them. This is the crux of the Triangle Removal lemma.
+-/
+
+open finset fintype
+
+namespace simple_graph
+variables {α 𝕜 : Type*} [linear_ordered_field 𝕜] (G H : simple_graph α) {n : ℕ} {s : finset α}
+
+section triangle_free
+
+/-- `G.triangle_free` means that `G` has no triangles. -/
+def triangle_free : Prop := ∀ t, ¬ G.is_n_clique 3 t
+
+variables {G H}
+
+lemma triangle_free_bot : (⊥ : simple_graph α).triangle_free :=
+begin
+  rintro t ⟨ht₁, ht₂⟩,
+  have : 1 < t.card,
+  { rw ht₁,
+    norm_num },
+  rw finset.one_lt_card at this,
+  obtain ⟨a, ha, b, hb, hab⟩ := this,
+  exact ht₂ ha hb hab,
+end
+
+lemma triangle_free.le (h : G ≤ H) : H.triangle_free → G.triangle_free :=
+forall_imp $ λ s, mt $ is_n_clique.le h
+
+end triangle_free
+
+section triangle_finset
+variables (G) [fintype α] [decidable_eq α] [decidable_rel G.adj] {a b c : α}
+
+/-- The triangles in a graph as a finset. -/
+def triangle_finset : finset (finset α) := (powerset_len 3 univ).filter $ G.is_n_clique 3
```

Comment:
For my application the finset is needed, but I think having a `set (finset` version could be useful for infinite graphs?

Reply:
I notice also that because we're filtering by `G.is_n_clique` anyway, the `powerset_len` isn't needed: we can just use `powerset`. And so the set version should be easy to write?